### PR TITLE
Fix for 416 and 417

### LINKED ISSUE #416: findOneAndReplace upsert not generating _id
When the `findOneAndReplace` API is called with `upsert` flag set to true, the new document is generated with the given properties in the `replacement` part of the request, but the `_id` is not generated.  In the below example, you could see `_id` and `upsertedId` are not set in the final response.

To replicate the issue;
```http
http: --- request POST http://localhost:8181/v1 
{
  "createNamespace": {
    "name": "testks1"
  }
}

http: --- request POST http://localhost:8181/v1/testks1 
{
  "createCollection": {
    "name": "products",
    "options": null
  }
}

http: --- request POST http://localhost:8181/v1/testks1/products {
  "findOneAndReplace": {
    "filter": {
      "name": "Product 25"
    },
    "replacement": {
      "name": "Product 25",
      "isCertified": false,
      "price": 20
    },
    "options": {
      "upsert": true,
      "returnDocument": "after"
    }
  }
}


http: --- response 200 POST http://localhost:8181/v1/testks1/products 
{
  "data": {
    "document": {
      "_id": null,
      "name": "Product 25",
      "isCertified": false,
      "price": 20
    }
  },
  "status": {
    "matchedCount": 0,
    "modifiedCount": 0,
    "upsertedId": null
  }
}

```

### PR DESCRIPTION
**Pull request by maheshrajamani**
**What this PR does**:
Fix for #416 and #417 
**Which issue(s) this PR fixes**:
Fixes #416 and #417 

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindOneAndReplaceIntegrationTest.java [hunk 1]
```diff
@@ -329,6 +330,52 @@ public void withUpsert() {
...
+        """;
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("data.documents[0]._id", is(notNullValue()));
```
Comment (vkarpov15) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindOneAndReplaceIntegrationTest.java:376]: Might be also good to check that the document's `_id` is a string

**Review by vkarpov15 — APPROVED** [review]
One minor comment but LGTM overall

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindOneAndReplaceIntegrationTest.java (hunk 1 above)
Comment (maheshrajamani) [reply src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindOneAndReplaceIntegrationTest.java:376]: Will do

**Comment by maheshrajamani** [comment]
Closing this for a new PR.

**Pull request closed without merging**